src/zarr/api/asynchronous.py

```diff
@@ -400,7 +401,9 @@ async def tree(*args: Any, **kwargs: Any) -> None:
     raise NotImplementedError
 
 
-async def array(data: npt.ArrayLike, **kwargs: Any) -> AsyncArray:
+async def array(
```

Comment:
For the future, but we can narrow the return type to a specific version when `zarr_format=2` or `zarr_format=3` is specified (either using overloads, or maybe a [type gurad](https://mypy.readthedocs.io/en/stable/type_narrowing.html#user-defined-type-guards).